dhall/src/semantics/resolve/resolve.rs

```diff
@@ -303,6 +322,51 @@ pub(crate) fn download_http_text(_url: Url) -> Result<String, Error> {
     panic!("Remote imports are not supported on wasm yet")
 }
 
+#[cfg(all(not(target_arch = "wasm32"), feature = "reqwest"))]
+pub(crate) fn cors_check(
+    origin: Option<String>,
+    remote: &Url,
+) -> Result<(), Error> {
+    if origin.is_none() {
+        return Ok(());
+    }
+    let origin = origin.unwrap();
+    let remote_origin =
+        format!("{}://{}", remote.scheme(), remote.host_str().unwrap());
+    if origin == remote_origin {
+        return Ok(());
+    }
+    let req = reqwest::blocking::Client::new()
+        .head(&remote.to_string())
+        .header(reqwest::header::USER_AGENT, "dhall-rust")
+        .header(reqwest::header::ORIGIN, &origin);
+
+    let resp = req.send().unwrap();
+    let headers = resp.headers();
+    if headers[reqwest::header::ACCESS_CONTROL_ALLOW_ORIGIN] != "*"
+        && headers[reqwest::header::ACCESS_CONTROL_ALLOW_ORIGIN] != origin
```

Comment:
This can panic if the `ACCESS_CONTROL_ALLOW_ORIGIN` header is not present. You should handle that case